services/worker/src/worker/utils.py

```diff
@@ -372,3 +378,35 @@ def is_dataset_script_error() -> bool:
     (t, v, tb) = sys.exc_info()
     cause_traceback: list[str] = traceback.format_exception(t, v, tb)
     return any(EXTERNAL_DATASET_SCRIPT_PATTERN in cause for cause in cause_traceback)
+
+
+def disable_dataset_scripts_support(allow_list: list[str]) -> AbstractContextManager[Any]:
+    original_init = HubDatasetModuleFactoryWithScript.__init__
+
+    def raise_unsupported_dataset_with_script_or_init(
+        self: HubDatasetModuleFactoryWithScript,
+        name: str,
+        revision: Optional[Union[str, datasets.Version]] = None,
+        download_config: Optional[DownloadConfig] = None,
+        download_mode: Optional[Union[datasets.DownloadMode, str]] = None,
+        dynamic_modules_path: Optional[str] = None,
+    ) -> None:
+        for allowed_pattern in allow_list:
+            if (allowed_pattern == "canonical" and "/" not in name) or fnmatch(name, allowed_pattern):
```

Comment:
maybe a more specific keyword instead of "canonical", as this org exists? (https://huggingface.co/Canonical)

even if it's handled by your code, it would feel less bug-prone.

You can use a forbidden character for example (see https://github.com/huggingface/moon-landing/blob/main/server/lib/Names.ts - internal)

Reply:
renamed `{{ALL_DATASETS_WITH_NO_NAMESPACE}}`